### ISSUE TITLE
Snappy/Lz4 should derive Deserialize and Serialize

### ISSUE THREAD

**Issue by ashneverdawn** [bonsairobo/building-blocks#11]
I am trying to switch to Snappy (to run as wasm). I have:
```
      let map: SerializableChunkMap<[i32; 3], f32, (), Snappy> =
        bincode::deserialize(&terrain_res.data[..]).unwrap();
```

Am getting this error:
```
error[E0277]: the trait bound `building_blocks::prelude::Snappy: Deserialize<'_>` is not satisfied
   --> src\terrain\terrain.rs:128:9
    |
128 |         bincode::deserialize(&terrain_res.data[..]).unwrap();
    |         ^^^^^^^^^^^^^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `building_blocks::prelude::Snappy`
```

Edit: Tested with Lz4 and same issue occurs as well.

**Comment by bonsairobo** [comment]
This is fixed on the main branch. I'll wait til making a release to close this issue, in case other people run into it.

**Comment by bonsairobo** [comment]
Fix is in v0.4
